core/src/main/scala/pureconfig/DerivedWriters.scala

```diff
@@ -18,14 +18,25 @@ trait DerivedWriters extends DerivedWriters1 {
     new ConfigWriter[T] {
       override def to(t: T): ConfigValue = writer.to(unwrapped.unwrap(t))
     }
+
+  // used for tuples
+  implicit final def deriveTupleInstance[F: IsTuple, Repr](
+    implicit
+    gen: Generic.Aux[F, Repr],
+    cc: ConfigWriter[Repr]): ConfigWriter[F] = new ConfigWriter[F] {
+    override def to(t: F): ConfigValue =
+      cc.to(gen.to(t))
+  }
 }
 
 /**
  * Trait containing `ConfigWriter` instances for collection, product and coproduct types.
  */
 trait DerivedWriters1 {
 
-  private[pureconfig] trait WrappedConfigWriter[Wrapped, SubRepr] extends ConfigWriter[SubRepr]
+  private[pureconfig] trait WrappedConfigWriter[Wrapped, SubRepr] {
+    def to(cv: SubRepr): ConfigValue
+  }
 
   implicit final def hNilConfigWriter[Wrapped]: WrappedConfigWriter[Wrapped, HNil] = new WrappedConfigWriter[Wrapped, HNil] {
```

Comment:
We should rename this and `hConsConfigWriter` to `labelledHNilConfigWriter` and `labelledHConsConfigWriter` in order to clearly distinguish them from the new ones. The same thing applies for readers.